crates/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -780,6 +783,26 @@ impl<'f> Context<'f> {
                     );
                     Instruction::Constrain(eq)
                 }
+                Instruction::Call { func, arguments } if func == assert_eq_id => {
+                    // Replace constraint `lhs == rhs` with `condition * lhs == condition * rhs`.
```

Comment:
Hmm, it seems like somewhat of a lateral move. I suppose it depends on whether `-` is generally faster than `*`. I think we can leave that to a separate PR though since it is mostly unrelated to assert_eq.